components/application-mgt/org.wso2.carbon.application.mgt.ui/src/main/java/org/wso2/carbon/application/mgt/ui/ApplicationAdminClient.java

```diff
@@ -89,22 +90,42 @@ public void deleteApp(String appNameWithVersion) throws AxisFault {
         }
         return null;
     }
-    public String getFaultException(String faultyAppName) throws AxisFault{
+
+    public String getFaultException(String faultyAppName) throws AxisFault {
         try {
             return stub.getFaultException(faultyAppName);
         } catch (java.lang.Exception e) {
             handleException(bundle.getString("cannot.get.service.data"), e);
         }
         return null;
     }
-    public void deleteFaultyApp(String[] faultyAppName) throws AxisFault{
+
+    public void deleteFaultyApp(String[] faultyAppName) throws AxisFault {
         try {
             stub.deleteFaultyApplication(faultyAppName);
         } catch (java.lang.Exception e) {
             handleException(bundle.getString("cannot.get.service.data"), e);
         }
     }
 
+    public void redeployApplication(String appName) throws AxisFault {
+        try {
+            stub.redeployApplication(appName);
+        } catch (java.lang.Exception e) {
+            handleException(bundle.getString("cannot.get.service.data"), e);
+        }
+    }
+
+    public void redeployApplications(String[] appNames) throws AxisFault {
+        try {
+            for (String appName: appNames) {
+                stub.redeployApplication(appName);
+            }
+        } catch (java.lang.Exception e) {
```

Comment:
Having the try outside the for loop will interrupt the redeployment of the rest of the Capps in the failure of a single Capp. Should we not try to redeploy the rest?

Reply:
Fixed in commit 7696d25